### PR TITLE
Fix never called BlockContextManager::setDefaultSettings on children

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Fixes https://github.com/sonata-project/SonataBlockBundle/issues/207
Fixes https://github.com/sonata-project/SonataPageBundle/pull/499

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
We need a tag after that.

File: Block/BlockContextManager.php [hunk 1]
```diff
@@ -139,7 +139,9 @@ public function get($meta, array $settings = array())
      */
     protected function setDefaultSettings(OptionsResolverInterface $optionsResolver, BlockInterface $block)
     {
-        trigger_error(__CLASS__.'::'.__METHOD__.' is deprecated since version 2.3, to be renamed in 3.0. Use '.__CLASS__.'::configureSettings instead.');
+        if (get_called_class() !== __CLASS__) {
```
Comment (rande) [inline Block/BlockContextManager.php]: GG

**Comment by Soullivaneuh** [comment]
https://github.com/sonata-project/SonataBlockBundle/releases/tag/2.3.3

File: Block/BlockContextManager.php (hunk 1 above)
Comment (Soullivaneuh) [inline Block/BlockContextManager.php]: Thanks! :-)

**Pull request merged**